apps/01PigLatin.js

```diff
@@ -4,30 +4,86 @@ var assert = require('assert');
 var prompt = require('prompt');
 prompt.start();
 
-//What is the significance of setting vowelIndex to -1? The entire array is search from back to front and when negative then it's taken from the end of the array as the offset. If calculated index is less than zero then it'll return -1 and the end of the search in the array.
 
 function pigLatin(word) {
 
-    word = word.toLowerCase();
-      var vowelIndex = -1;
-
-      if (( word.indexOf('a') > -1 && word.indexOf('a') < vowelIndex) || vowelIndex === -1) {
-         vowelIndex = word.indexOf('a');
-      }
-      if (( word.indexOf('e') > -1 && word.indexOf('e') < vowelIndex) || vowelIndex === -1) {
-         vowelIndex = word.indexOf('e');
-      }
-      if (( word.indexOf('i') > -1 && word.indexOf('i') < vowelIndex) || vowelIndex === -1) {
-         vowelIndex = word.indexOf('i');
-      }
-      if (( word.indexOf('o') > -1 && word.indexOf('o') < vowelIndex) || vowelIndex === -1) {
-         vowelIndex = word.indexOf('o');
-      }
-      if (( word.indexOf('u') > -1 && word.indexOf('u') < vowelIndex) || vowelIndex === -1) {
-         vowelIndex = word.indexOf('u');
-      }
-}
+//Concept check
+//
+//Please explain why we are using String.prototype.toLowerCase() here.
+//For more info:
+//https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase
+//
+//Your Answer:
+//It is used to convert a string to lowercase letters, but it does not change the orginal string.
```

Comment:
It's used at the beginning so it converts cap letters used when starting the game. 

Sent from my iPhone

> On Aug 14, 2016, at 4:07 PM, maddiesolz <notifications@github.com> wrote:
> 
> In apps/01PigLatin.js:
> 
> > -      }
> > -      if (( word.indexOf('o') > -1 && word.indexOf('o') < vowelIndex) || vowelIndex === -1) {
> > -         vowelIndex = word.indexOf('o');
> > -      }
> > -      if (( word.indexOf('u') > -1 && word.indexOf('u') < vowelIndex) || vowelIndex === -1) {
> > -         vowelIndex = word.indexOf('u');
> > -      }
> > -}
> > +//Concept check
> > +//
> > +//Please explain why we are using String.prototype.toLowerCase() here.
> > +//For more info:
> > +//https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase
> > +//
> > +//Your Answer:
> > +//It is used to convert a string to lowercase letters, but it does not change the orginal string.
> Yes this is true, but why are we converting the string to lower case first before we start doing the code for the app?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>